src/Common/src/Interop/Windows/advapi32/Interop.ReportEvent.cs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+internal partial class Interop
+{
+    internal partial class Advapi32
+    {
+        [DllImport(Libraries.Advapi32, CharSet = CharSet.Unicode, SetLastError = true)]
+        public static extern bool ReportEvent(
+            SafeEventLogWriteHandle hEventLog,
+            short wType,
+            ushort wcategory,
+            uint dwEventID,
+            byte[] lpUserSid,
+            short wNumStrings,
+            int dwDataSize,
+            HandleRef lpStrings,
```

Comment:
Similar: HandleRef can be just `IntPtr` here and the callsite can directly pass the result of AddrOfPinnedObject in. `HandleRef` serves no purpose here.